src/pages/home/sidebar/SidebarLinksData.js

```diff
@@ -134,6 +123,23 @@ function SidebarLinksData({
     // eslint-disable-next-line react-hooks/exhaustive-deps
     useEffect(() => Policy.openWorkspace(activeWorkspaceID, policyMemberAccountIDs), [activeWorkspaceID]);
 
+    const reportIDsWithErrors = useMemo(() => {
+        const reportKeys = _.keys(chatReports);
+        return _.reduce(
+            reportKeys,
+            (errorsMap, reportKey) => {
+                const report = chatReports[reportKey];
+                const allReportsActions = allReportActions[reportKey.replace(ONYXKEYS.COLLECTION.REPORT, ONYXKEYS.COLLECTION.REPORT_ACTIONS)];
+                const errors = OptionsListUtils.getAllReportErrors(report, allReportsActions) || {};
```

Comment:
I think we should pass the violations to `getAllReportErrors` and add them to the dependencies of the memo. If we don't, this function will not be re-triggered if a violation changes in Onyx.

Reply:
But it is not a dependency of `reportIDsWithErrors` [here](https://github.com/Expensify/App/blob/1eca3e126b89bf7dd762447e26951654611bdbfa/src/pages/home/sidebar/SidebarLinksData.js#L141), which means we won't re-run `getAllReportErrors`

Reply:
While yes the page will re-render, the `reportIDsWithErrors` will remain unchanged, as it only listens to changes in report actions and chat reports.

Reply:
If a change is made to the violation only, that function will not be re-triggered, I think the video below from @shubham1206agra, it is most likely because changing the category and fixing the violation has made changes on the report action which triggered refreshing `reportIDsWithErrors `. In cases where there might be an update from the server that's only altering the violations, `reportIDsWithErrors` will not be refreshed as it's simply not listening to changes in the violations.

Reply:
Ahh apologies, I forgot `getAllReportErrors` does not include transaction violations!

Reply:
Ok but now we have an issue with the transaction not being memoized, we rely on it [here](https://github.com/Expensify/App/blob/1eca3e126b89bf7dd762447e26951654611bdbfa/src/libs/OptionsListUtils.ts#L489) to display smartscan errors and I think what I said above will apply to it. Can you guys check please? (I think my previous should have been about the transaction and not the violations)

Reply:
The updates coming from the backend include both the transaction and report action, but what if there's a race condition and when we re-render due to the report action being changed, the Onyx transaction update hasn't been applied yet? I think we should always include dependencies in the memo to avoid this scenario.

Reply:
Agreed

Reply:
@shubham1206agra bump, can we pass the transactions to `getAllReportErrors`?

Reply:
Apologies for the confusion, @fedirjh I disagree. Prior to the PR we weren’t pinning RBR reports, now that we’re memoized them, they are not subscribing to changes in the transaction.

Reply:
@shubham1206agra @fedirjh what's the latest here?